### ISSUE TITLE
GCC trunk not built for over a week due to GNAT build failure

### ISSUE THREAD

**Issue by mattgodbolt** [mattgodbolt/compiler-explorer-image#161]
```
../../../../../../../gcc-trunk-20190129/libstdc++-v3/src/filesystem/ops.cc: In function 'T {anonymous}::do_stat(const std::experimental::filesystem::v1::__cxx11::path&, std::error_code&, Accessor, T) [with Accessor = std::experimental::filesystem::v1::file_size(const std::experimental::filesystem::v1::__cxx11::path&, std::error_code&)::<lambda(const auto:1&)>; T = std::experimental::filesystem::v1::file_size(const std::experimental::filesystem::v1::__cxx11::path&, std::error_code&)::S]':
../../../../../../../gcc-trunk-20190129/libstdc++-v3/src/filesystem/ops.cc:783:5: warning: empty class 'std::experimental::filesystem::v1::file_size(const std::experimental::filesystem::v1::__cxx11::path&, std::error_code&)::<lambda(const auto:1&)>' parameter passing ABI changes in -fabi-version=12 (GCC 8) [-Wabi]
  783 |     do_stat(const fs::path& p, std::error_code& ec, Accessor f, T deflt)
      |     ^~~~~~~
```

Not sure if this is our issue or upstream...

**Comment by davidmalcolm** [comment]
FWIW I just did a clean build (r268362 from today, using gcc 4.8 as the bootstrap compiler) and I'm not seeing that warning.  (I note that that's a warning, not an error; are you manually failing the build if any warning occurs?)

Are the full logs from your build visible somewhere?

**Comment by jwakely** [comment]
It's just a warning, how is it preventing GCC from building?

**Comment by mattgodbolt** [comment]
Yes! The full logs are at https://godbolt.org/admin/builds.html (click the "build log" on the failing).

Good question both. Reading the error output _properly_ yields:

```
raise-gcc.c: In function 'setup_to_install':
raise-gcc.c:1120:4: warning: cast from pointer to integer of different size [-Wpointer-to-int-cast]
 1120 |    (_Unwind_Word)uw_exception);
      |    ^
g-debpoo.adb: In function 'GNAT.DEBUG_POOLS.GET_SIZE':
g-debpoo.adb:1440:8: warning: 'SIZE_IN_STORAGE_ELEMENTS' may be used uninitialized in this function [-Wmaybe-uninitialized]
g-debpoo.ads:295:7: note: 'SIZE_IN_STORAGE_ELEMENTS' was declared here
g-comlin.adb: In function 'GNAT.COMMAND_LINE.FIND_LONGEST_MATCHING_SWITCH':
g-comlin.adb:583:8: warning: 'PARAM' may be used uninitialized in this function [-Wmaybe-uninitialized]
g-comlin.adb:107:7: note: 'PARAM' was declared here
g-catiio.adb: In function 'GNAT.CALENDAR.TIME_IO.VALUE':
g-catiio.adb:98:8: warning: 'TIME' may be used uninitialized in this function [-Wmaybe-uninitialized]
g-catiio.adb:941:69: warning: 'LOCAL_DISP' may be used uninitialized in this function [-Wmaybe-uninitialized]
g-catiio.adb:830:7: note: 'LOCAL_DISP' was declared here
s-osinte.ads:120:41: "SIGSYS" not declared in "Linux"
make[9]: *** [a-dispat.o] Error 1
make[9]: *** Waiting for unfinished jobs....
s-osinte.ads:120:41: "SIGSYS" not declared in "Linux"
make[9]: *** [a-dynpri.o] Error 1
make[8]: *** [gnatlib] Error 2
make[7]: *** [gnatlib-shared-default] Error 2
make[6]: *** [gnatlib-shared-dual] Error 2
make[5]: *** [gnatlib-shared] Error 2
make[4]: *** [gnatlib-shared] Error 2
make[3]: *** [multi-do] Error 1
make[2]: *** [all] Error 2
make[1]: *** [all-target-libada] Error 2
make[1]: *** Waiting for unfinished jobs....
```

-- it's the ADA support that's failing! We can look to disable that for trunk.

**Comment by jwakely** [comment]
The -Wabi warning *is* an upstream issue, but is harmless and can be ignored.

**Comment by Mitchtdk** [comment]
In my testing and attempts at building GCC trunk over the past few days I seem to have been able to narrow down the issue with GNAT failing to build being due to the multilib support of x32.

I haven't had any issues building GCC 8.2 with current configuration CE uses, and I have been able to build & run GCC trunk (9.0.1 20190203), including GNAT if I remove the `mx32` as a parameter for the `--with-multilib-list` option. 

So now I'm thinking that something has broken since the release of 8.2 and the current trunk of GCC that affects GNAT, libada and support for x32.

**Comment by jwakely** [comment]
> So now I'm thinking that something has broken since the release of 8.2 and the current trunk

The fact that 8.2 works and trunk doesn't only tells you that something changed on trunk between the branch point for `gcc-8-branch` (April 2018) and now, not between the release of 8.2 (July 2018) and now.

**Comment by Mitchtdk** [comment]
Yeah sorry your right, release was the wrong word, I should have clarified that I meant the 8.2 branch.

**Comment by Mitchtdk** [comment]
I sent an email to AdaCore yesterday via their bug reporting email, and I received a patch to apply today. I have confirmed with them that on my VM I have been able to successfully build GCC (including GNAT) with the default multilib-list options CE has (`--with-multilib-list=m32,m64,mx32`). They have confirmed that the patch has been up streamed. 

I have rebuilt the updated trunk and again I have not had anymore issues in building GCC (`gcc (GCC-Explorer-Build) 9.0.1 20190206 (experimental)`) and GNAT (`GNAT 9.0.1 20190206 (experimental)`).
